README.md

```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
 and reproduce Storage API transient errors. The testbench emulates the Storage API and
 is expected to be used by Storage library maintainers.
 
+
+## Table of Contents
+* [Issue Policy](#issue-policy)
+* [What is this testbench/emulator?](#what-is-this-emulator)
+* [When to use](#when-to-use-this-emulator)
+* [How do I use this emulator?](#how-to-use-this-emulator)
+  - [Initial set up](#initial-set-up)
+  - [Run the emulator](#run-the-emulator)
+  - [Check that the emulator is running](#check-that-the-emulator-is-running)   
+
 ## Issue Policy
 
 Repository provides no dedicated support for issues filed.
 Issues will be addressed when time permits.
+
+## What is this emulator?
+**Note that the words emulator and testbench are used interchangeably below.
+
+This emulator fakes the Google Cloud Storage (GCS) API. You can configure the GCS client libraries to make calls to this fake rather than to the actual API.
+* The emulator fakes the JSON API (both over REST and gRPC), and has limited support for the XML API.
+* Generally, the error codes are similar to the ones generated by GCS, but the error messages are not. 
+* The emulator performs far fewer error checks and no permission checks (ACL/IAM). 
+
+## When to use this emulator
+In general, this emulator is best suited for integration tests that are hard or even annoying to reliably run against production. The primary example of this are errors that make the client library go through its retry path.
+
+This emulator can be useful to test HMAC keys, which are really hard to test against production due to quota restrictions.
+
+It is useful as well to test features that are not yet deployed to production, you can implement them in the emulator and then write the library code before production is "ready".
+
+## How to use this emulator
+
+### Initial set up
+
+1. [Set up python if you haven't already](https://cloud.google.com/python/docs/setup)
+2. Clone this repository
+3. ```bash
```

Comment:
Sometimes you describe what to do and the command to do it, sometimes just the command, sometimes what do to without command.  Maybe consistently do both?